### PR TITLE
perfdash: update to aws-sdk-v2 for Go

### LINKED ISSUE #2796: switch to aws-sdk-go-v2 as v1 is going away
Context: https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-go-v1-on-july-31-2025/
Migration guide: https://aws.github.io/aws-sdk-go-v2/docs/migrating/

Basically maintenance mode on July 31, 2024 and reach end-of-support on July 31, 2025. The SDK will not 
receive API updates for new or existing services, or be updated to support new regions.

### PR DESCRIPTION
**Pull request by mboersma**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Updates S3 usage in perfdash to use aws-sdk-v2 for Go, since v1 is going into maintenance mode and will be unsupported a year from now.

#### Which issue(s) this PR fixes:

Fixes #2796

#### Special notes for your reviewer:

See the migration guide at: https://aws.github.io/aws-sdk-go-v2/docs/migrating/

I think this is the only usage of aws-sdk in the project. The only other refs are indirect in a `go.sum` file, not called by this code.

### CODE REVIEW

**Comment by mboersma** [comment]
/retitle [WIP] perfdash: update to aws-sdk-v2 for Go

Oops, looks like more refactoring is needed. I'll take a look..

File: perfdash/go.mod [hunk 1]
```diff
@@ -16,11 +15,33 @@ require (
 	k8s.io/kubernetes v1.15.0
 )
 
+require (
```
Comment (wojtek-t) [inline perfdash/go.mod:18]: nit: why do we need separate require section for it?

Comment (mboersma) [reply perfdash/go.mod]: We don't, but `go mod tidy` chose to do so for some reason. I added the imports to an existing section and `go mod tidy` seems fine with that.

File: perfdash/s3_metrics_bucket.go [hunk 2]
```diff
@@ -17,32 +17,36 @@ limitations under the License.
...
+	cfg, err := awsconfig.LoadDefaultConfig(context.Background(),
+		awsconfig.WithRegion(region),
+	)
+	if err != nil {
+		return nil, err
+	}
 	return &S3MetricsBucket{
-		client: s3.New(session.Must(session.NewSession(&aws.Config{
-			Region:                        aws.String(region),
-			CredentialsChainVerboseErrors: aws.Bool(true),
```
Comment (mboersma) [inline perfdash/s3_metrics_bucket.go:43]: I couldn't find an SDK v2 replacement for `CredentialsChainVerboseErrors`, so I left it out.

**Comment by mboersma** [comment]
/retitle perfdash: update to aws-sdk-v2 for Go

**Pull request merged**